WordPress/Classes/Utility/BackgroundTasks/WeeklyRoundupBackgroundTask.swift

```diff
@@ -538,21 +537,15 @@ class WeeklyRoundupNotificationScheduler {
         comments: Int,
         likes: Int,
         periodEndDate: Date,
-        context: NSManagedObjectContext,
         completion: @escaping (Result<Void, Error>) -> Void) {
 
-            var siteTitle: String?
-            var dotComID: Int?
+        let siteTitle = site.title
+        let dotComID = site.dotComID?.intValue
 
-            context.performAndWait {
```

Comment:
I'm pretty sure it's safest to ensure that these operations accessing the object happen on the same queue as its context. This change was added because originally we were setting crashes here when accessing `siteTitle`. If I recall we didn't ever really get to the bottom of what was causing the issue but I think this change improved things (can't say for certain though). I'd personally be hesitant to change it. Can we just call perform on the site's `context` if we don't want to pass one in?